tests/check_check_tags.c

```diff
@@ -0,0 +1,621 @@
+/*
+ * Check: a unit test framework for C
+ * Copyright (C) 2001, 2002 Arien Malec
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+#include "../lib/libcompat.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <check.h>
+#include <assert.h>
+#include "check_check.h"
+
+enum {
+    RED,
+    BLUE,
+    PURPLE,
+    YELLOW,
+    BLACK,
+    MAX_TESTS
+} test_ids;
+
+unsigned int test_executed[MAX_TESTS];
+
+static void reset_executed(void)
+{
+    unsigned int i;
+
+    for (i = 0; i < MAX_TESTS; i++)
+    {
+	test_executed[i] = 0;
+    }
+}
+
+START_TEST(red_test1)
+{
+    test_executed[RED] = 1;
+}
+END_TEST
+
+START_TEST(blue_test1)
+{
+    test_executed[BLUE] = 1;
+}
+END_TEST
+
+START_TEST(purple_test1)
+{
+    test_executed[PURPLE] = 1;
+}
+END_TEST
+
+START_TEST(yellow_test1)
+{
+    test_executed[YELLOW] = 1;
+}
+END_TEST
+
+START_TEST(black_test1)
+{
+    test_executed[BLACK] = 1;
+}
+END_TEST
+
+
+static Suite *make_tagged_suite(void)
+{
+    Suite *s;
+
+    TCase *red, *blue, *purple, *yellow, *black;
+
+    s = suite_create("Check Tag Filtering");
+
+    red = tcase_create("Red");
```

Comment:
nitpick, but why not initialize the Red tests in the same was as the other tests? Namely, 4 lines together without blank lines?